strutil/strutil.go

```diff
@@ -209,3 +209,19 @@ func ElliptRight(str string, n int) string {
 	// this is expensive; look into a cheaper way maybe sometime
 	return string([]rune(str)[:n-1]) + "…"
 }
+
+// ElliptLeft returns a string that is at most n runes long,
+// replacing the first rune with an ellipsis if necessary. If N is less
+// than 1 it's treated as a 1.
+func ElliptLeft(str string, n int) string {
+	if n < 1 {
+		n = 1
+	}
+	// this is expensive; look into a cheaper way maybe sometime
+	rstr := []rune(str)
```

Comment:
Maybe checking `utf8.RuneCountInstring(str) < n` before making a `[]rune` would be lest expensive. Or maybe not? The cost is probably negligible given where it's called.

Reply:
I need to measure (note `ElliptRight` does do that), but it's essentially doing the decode twice.